mpmp/utilities/file_utilities.py

```diff
@@ -16,56 +16,47 @@ def make_output_dir(experiment_dir, identifier):
 
 
 def construct_filename(output_dir,
-                       identifier,
-                       shuffle_labels,
-                       model_options,
                        file_descriptor,
-                       extension):
-    """Construct a filename from experimental parameters.
-
-    This allows us to standardize what information goes into the filename
-    (i.e. what information constitutes a unique "experiment").
+                       extension,
+                       *args,
+                       **kwargs):
+    """Construct a filename from varying experimental parameters.
+
+    The format of the filename will roughly look like this:
+    {output_dir}/{a}_{k}_{file_descriptor}{extension}
+    where a = '_'.join([ar for ar in args]),
+          k = '_'.join(['{k}{v} for k, v in kwargs.items()])
+
+    For example:
+    >>> construct_filename('output_dir', 'output', '.tsv',
+    ...                    'expression', 'signal',
+    ...                    s=42, n=5000)
+    output_dir/expression_signal_s42_n5000_output.tsv
+
+    Also note that if a parameter has a value of None, it will be silently
+    skipped in the output filename.
     """
-    training_data = model_options.training_data
-    signal = 'shuffled' if shuffle_labels else 'signal'
-    seed = model_options.seed
-    if identifier is None:
-        # this might be the case for files that pertain to the whole experiment
-        # (e.g. metadata/parameters file)
-        try:
-            return Path(output_dir,
-                        '{}_s{}_n{}_{}{}'.format(training_data,
-                                                 seed,
-                                                 model_options.n_dim,
-                                                 file_descriptor,
-                                                 extension))
-        except AttributeError:
-            # no n_dim in model_options => not a compressed model
-            return Path(output_dir,
-                        '{}_s{}_{}{}'.format(training_data,
-                                             seed,
-                                             file_descriptor,
-                                             extension))
+    if len(args) == 0 and len(kwargs) == 0:
+        return Path(output_dir,
+                    '{}{}'.format(file_descriptor, extension))
```

Comment:
In case you want (and can) use f-strings.

```suggestion
                    f'{file_descriptor}{extension}'
```

Reply:
I should really update all of my code to use f-strings (I have issue #12 to go through and do that). I'm planning to take a few days to do some code cleanup in the near future, so I'll try to roll this into those changes.